internal/kafka/internal/services/kafka.go

```diff
@@ -1150,6 +1214,73 @@ func buildManagedKafkaCR(kafkaRequest *dbapi.KafkaRequest, kafkaConfig *config.K
 	return managedKafkaCR, nil
 }
 
+// buildReservedManagedKafkaCR buklds a Reserved Managed Kafka CR.
+// The ID, K8s object ID, K8s namespace and PlacementID are all set to
+// the provided kafkaID.
+func (k *kafkaService) buildReservedManagedKafkaCR(kafkaID string, instanceType string, desiredStrimziVersion api.StrimziVersion, kafkaConfig *config.KafkaConfig) (*managedkafka.ManagedKafka, *errors.ServiceError) {
+	// Reserved instances always make use of the streaming base unit which is x1.
+	// For now we hardcode it but there might be a better alternative to it.
+	streamingBaseUnit := "x1"
+	kafkaInstanceSize, err := kafkaConfig.GetKafkaInstanceSize(instanceType, streamingBaseUnit)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "unable to list kafka request")
+	}
+	labels := map[string]string{
+		"bf2.org/kafkaInstanceProfileQuotaConsumed": strconv.Itoa(kafkaInstanceSize.QuotaConsumed),
+		"bf2.org/kafkaInstanceProfileType":          instanceType,
+		"bf2.org/deployment":                        "reserved",
```

Comment:
shall we make `reserved` into a constant?

Reply:
:+1: . I thought about it but I didn't do it. It does no harm so I added it as a constant and I also added the label key as a constant